### PR TITLE
Stop the config file from overwriting an explicit --org

### PR DESCRIPTION
**Pull request by no-itsbackpack**
## Summary

`pscale <anything> --org <org>` silently ignored `--org` when `~/.config/planetscale/pscale.yml` set an org, running against the configured organization instead. It is on `main` today, so agents and scripts that pass `--org` on every command (as `AGENTS.md` instructs) would read and write the wrong organization without any warning.

Cause: `initConfig` walked the whole command tree and preset every command's flags from viper. Commands bind `--org` to the same `ch.Config.Organization` field, so presetting an unrelated command's unset `--org` wrote the config value over the one parsed from the command line. This surfaced once a command declared `--org` as a local (not persistent) flag, since only local flags are in `cmd.Flags()` for commands cobra never merges persistent flags into.

Fix: preset only the command being run, and skip flags that were set on the command line.

Also points the CLI tests at an empty home directory. They were reading the developer's real config and credentials, which made `TestRootSkillFlagDoesNotInterceptSubcommandValues` fail locally (it reached the API instead of `NO_AUTH`).

## Test plan

- [x] `go test ./...`
- [x] New tests: `--org` reaches the API request, config still fills an unset `--org`, and `targetCommand` resolves subcommands and falls back to root
- [x] Verified against production with a local build:
  - `pscale database list --org does-not-exist-xyz --format json` errors instead of listing the configured org's databases
  - `pscale database list --format json` still uses the org from the config file
  - `pscale database list --org example --format json` lists that org
  - `pscale org update --billing-email ... --api-url http://127.0.0.1:1/` still fills its required `--org` from the config file (request path is `/v1/organizations/<config org>`)

### CODE REVIEW

**Review by nholden — APPROVED** [review]

File: internal/cmd/root.go [hunk 1]
```diff
@@ -488,27 +488,35 @@ func initConfig() {
...
-			postInitCommands(cmd.Commands())
-		}
+// targetCommand returns the command cobra is about to run, falling back to the
+// root command when the arguments don't resolve (unknown command, no args).
+func targetCommand(root *cobra.Command, args []string) *cobra.Command {
+	cmd, _, err := root.Find(args)
+	if err != nil || cmd == nil {
+		return root
 	}
+	return cmd
```
[the remark this answers is not in the document]
Comment (no-itsbackpack) [reply internal/cmd/root.go:503]: Fixed in b17c671b. Target selection now uses Cobra’s Traverse path, matching Execute when TraverseChildren is enabled. Added coverage for `database --org acme list` in both target resolution and the config-precedence integration test.
